src/core/r3000a.cc

```diff
@@ -313,37 +313,37 @@ void PCSX::R3000Acpu::branchTest() {
     uint32_t* targets = m_regs.intTargets;
 
     if ((interrupts != 0) && (((int32_t)(m_regs.lowestTarget - cycle)) <= 0)) {
-        auto checkAndUpdate = [&lowestDistance, &lowestTarget, interrupts, cycle, targets, this](
-                                  unsigned interrupt, std::function<void()> act) {
-            uint32_t mask = 1 << interrupt;
-            if ((interrupts & mask) == 0) return;
-            uint32_t target = targets[interrupt];
-            int32_t dist = target - cycle;
-            if (dist > 0) {
-                if (lowestDistance > dist) {
-                    lowestDistance = dist;
-                    lowestTarget = target;
-                }
-            } else {
-                m_regs.interrupt &= ~mask;
-                PSXIRQ_LOG("Triggering interrupt %08x\n", interrupt);
-                act();
-            }
-        };
-
-        checkAndUpdate(PSXINT_SIO, []() { g_emulator->m_sio->interrupt(); });
-        checkAndUpdate(PSXINT_SIO1, []() { g_emulator->m_sio1->interrupt(); });
-        checkAndUpdate(PSXINT_CDR, []() { g_emulator->m_cdrom->interrupt(); });
-        checkAndUpdate(PSXINT_CDREAD, []() { g_emulator->m_cdrom->readInterrupt(); });
-        checkAndUpdate(PSXINT_GPUDMA, []() { GPU::gpuInterrupt(); });
-        checkAndUpdate(PSXINT_MDECOUTDMA, []() { g_emulator->m_mdec->mdec1Interrupt(); });
-        checkAndUpdate(PSXINT_SPUDMA, []() { spuInterrupt(); });
-        checkAndUpdate(PSXINT_MDECINDMA, []() { g_emulator->m_mdec->mdec0Interrupt(); });
-        checkAndUpdate(PSXINT_GPUOTCDMA, []() { gpuotcInterrupt(); });
-        checkAndUpdate(PSXINT_CDRDMA, []() { g_emulator->m_cdrom->dmaInterrupt(); });
-        checkAndUpdate(PSXINT_CDRPLAY, []() { g_emulator->m_cdrom->playInterrupt(); });
-        checkAndUpdate(PSXINT_CDRDBUF, []() { g_emulator->m_cdrom->decodedBufferInterrupt(); });
-        checkAndUpdate(PSXINT_CDRLID, []() { g_emulator->m_cdrom->lidSeekInterrupt(); });
+#define checkAndUpdate(irq, act)                                \
```

Comment:
I'm not convinced the codegen here is better, but, okay.

This sort of lambda should be expanded as-is by the compiler, (especially now that we're on clang everywhere for Release), and have in effect the same result as the macro. If it's not, something else is weird and would have to be investigated.